src/AzureAuth/IEnvExtensions.cs

```diff
@@ -22,5 +26,38 @@ public static bool InteractiveAuthDisabled(this IEnv env)
             return !string.IsNullOrEmpty(env.Get(EnvVars.NoUser)) ||
                 string.Equals(CorextPositiveValue, env.Get(EnvVars.CorextNonInteractive));
         }
+
+        /// <summary>
+        /// Get the auth modes from the environment or set the default.
+        /// </summary>
+        /// <param name="env">The <see cref="IEnv"/> to use.</param>
+        /// <param name="eventData">Event data to add the auth mode to.</param>
+        /// <returns>AuthModes.</returns>
+        public static IEnumerable<AuthMode> ReadAuthModeFromEnvOrSetDefault(this IEnv env, EventData eventData)

```

Comment:
nit: Is there an alternative to passing in eventData here? It does seem a bit out of place for an environment extension method, but I can understand the desire to centralize telemetry.

Reply:
Also, feel free to resolve this comment whenever since I won't be available to review next week. Just wanted to raise a thought 😄

Reply:
That sounds right. I just found a way to put this env variables under the existing env variables we collect for telemetry here https://github.com/AzureAD/microsoft-authentication-cli/blob/a151ef0a7648b89f47d58ced2eaf08cccf77e9c2/src/AzureAuth/Program.cs#L65

Reply:
Done